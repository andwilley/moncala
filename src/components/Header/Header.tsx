import React from "react";

export const Header: React.FC = () => {
  return (
    <div className="Header h3 pt2 pl2">
      <div className="dib">
        <h2 className="ma0 b">Of Note</h2>
      </div>
      <div className="dib fr h2 w2 mr2 ba">add</div>
    </div>
  );
};

export default Header;
