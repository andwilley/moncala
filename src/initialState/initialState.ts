import { IState } from '../types/IState';
import { EditorState } from 'draft-js';

export const initialState: IState = {
  notes: {
    byId: {},
    allIds: []
  },
  activeId: undefined,
  editorState: EditorState.createEmpty(),
  focusEditor: false
};
