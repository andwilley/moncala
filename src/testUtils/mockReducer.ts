import { IState } from '../types/IState';
import { Action } from '../actions/actions';

const mockReducer = (testState: IState) => (
  state: IState | undefined = testState,
  action: Action
) => state;

export default mockReducer;
