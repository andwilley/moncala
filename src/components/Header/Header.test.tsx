import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import mockReducer from '../../testUtils/mockReducer';
import { initialState } from '../../initialState/initialState';

describe('<Header> test a basic render output', () => {
  const store = createStore(mockReducer(initialState));
  it('should render the notepad', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
