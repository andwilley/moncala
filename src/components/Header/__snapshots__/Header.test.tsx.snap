// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> test a basic render output should render the notepad 1`] = `
<div
  className="Header h3 pt2 pl2"
>
  <div
    className="dib w5"
  >
    <h2
      className="ma0 b"
    >
      Of Note
    </h2>
  </div>
  <button
    className="ba bg-transparent pr2 dim f6 gray ba b--moon-gray pointer br2 "
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="flex items-center"
    >
      <span
        className="dib h2 w2 pointer"
        onClick={[Function]}
      >
        <svg
          className=""
          height={30}
          style={
            Object {
              "fill": "#555555",
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      New Note
    </span>
  </button>
</div>
`;
